### PR TITLE
Update to windows 0.58

### PR DESCRIPTION
**Pull request by Jasper-Bekkers**
Not a huge change, this updates `wmi-rs` to `windows 0.58`

### CODE REVIEW

File: src/safearray.rs [hunk 1]
```diff
@@ -4,6 +4,7 @@ use crate::{
 };
 use std::{iter::Iterator, ptr::null_mut, slice};
 use windows::core::BSTR;
+// use windows_core::imp::SAFEARRAY;
```
Comment (ohadravid) [inline src/safearray.rs:7]: Remove?

File: src/variant.rs [hunk 2]
```diff
@@ -3,10 +3,11 @@ use crate::{
 };
 use serde::Serialize;
 use std::convert::TryFrom;
-use windows::core::{ComInterface, IUnknown, BSTR};
-use windows::Win32::Foundation::{VARIANT_FALSE, VARIANT_TRUE};
+use windows::core::{IUnknown, BSTR, VARIANT};
+use windows::Win32::Foundation::{VARIANT_BOOL, VARIANT_FALSE, VARIANT_TRUE};
 use windows::Win32::System::Variant::*;
 use windows::Win32::System::Wmi::{self, IWbemClassObject, CIMTYPE_ENUMERATION};
+use windows_core::Interface;
```
Comment (ohadravid) [inline src/variant.rs:10]: Can this be `use windows::core::Interface;`, as part of line 6?

File: src/variant.rs [hunk 3]
```diff
@@ -165,11 +169,11 @@ impl Variant {
             VT_EMPTY => Variant::Empty,
             VT_NULL => Variant::Null,
             VT_UNKNOWN => {
-                let unk = unsafe { &vt.Anonymous.Anonymous.Anonymous.punkVal };
-                let ptr = unk.as_ref().ok_or(WMIError::NullPointerResult)?;
+                let unk = unsafe { &(vt.Anonymous.Anonymous.Anonymous.punkVal as *const IUnknown) };
+                let ptr = unsafe { unk.as_ref() }.ok_or(WMIError::NullPointerResult)?;
```
Comment (ohadravid) [inline src/variant.rs:173]: I think both lines can be simplified to 
`let ptr = unsafe { IUnknown::from_raw(vt.Anonymous.Anonymous.Anonymous.punkVal) };`

File: src/connection.rs [hunk 4]
```diff
@@ -63,7 +63,7 @@ impl COMLibrary {
     /// `CoInitialize`s the COM library for use by the calling thread, but without setting the security context.
     ///
     pub fn without_security() -> WMIResult<Self> {
-        unsafe { CoInitializeEx(None, COINIT_MULTITHREADED)? }
+        unsafe { CoInitializeEx(None, COINIT_MULTITHREADED).ok()? }
```
Comment (ohadravid) [inline src/connection.rs:66]: This will lose the original error code - would you mind adding the needed variant to the WMIError?

**Review by ohadravid — COMMENTED** [review]
Thanks for taking the time to work on this 🎉

File: src/connection.rs (hunk 4 above)
Comment (Jasper-Bekkers) [reply src/connection.rs:66]: It's slightly misleading but this calls https://docs.rs/windows-core/latest/windows_core/struct.HRESULT.html#method.ok which does retain the error code.

**Comment by Jasper-Bekkers** [comment]
Addressed all feedback, this should (tm) make the tests run again as well.

File: src/connection.rs (hunk 4 above)
Comment (ohadravid) [reply src/connection.rs:66]: 🤯

**Comment by Jasper-Bekkers** [comment]
> LGTM 🎉

Would you mind spinning a release if/when this lands?

**Comment by ohadravid** [comment]
Oh no:

```
> cargo test -- --nocapture --test-threads 1 it_can_desr_newtype_enum_field
...

Caused by:
  process didn't exit successfully: `... --nocapture --test-threads 1 _enum_fi` (exit code: 0xc0000005, STATUS_ACCESS_VIOLATION)
```

**Comment by Jasper-Bekkers** [comment]
My first guess would be a problem I've seen before with the `implement` macro but this requires a bit more investigation.

**Comment by ohadravid** [comment]
@Jasper-Bekkers - seems like an issue with `IUnknown::from_raw`, which should be the `_borrowed` version.

Can you merge the changes from my branch (https://github.com/ohadravid/wmi-rs/tree/update-win-0.58-fixes) into yours?

**Comment by Jasper-Bekkers** [comment]
> @Jasper-Bekkers - seems like an issue with `IUnknown::from_raw`, which should be the `_borrowed` version.
> 
> Can you merge the changes from my branch (https://github.com/ohadravid/wmi-rs/tree/update-win-0.58-fixes) into yours?

Done - looks like it needed a cargo fmt still, so I pushed that. Thanks for taking the time to look into this.

File: src/variant.rs [hunk 5]
```diff
@@ -97,11 +101,13 @@ impl Variant {
         // See https://msdn.microsoft.com/en-us/library/cc237865.aspx for more info.
         // Rust can infer the return type of `vt.*Val()` calls,
         // but it's easier to read when the type is named explicitly.
-        let variant_value = match variant_type {
+        let variant_value = match VARENUM(variant_type) {
             VT_BSTR => {
-                let bstr_ptr: &BSTR = unsafe { &vt.Anonymous.Anonymous.Anonymous.bstrVal };
-
-                Variant::String(bstr_ptr.try_into()?)
+                let bstr_ptr = unsafe { BSTR::from_raw(vt.Anonymous.Anonymous.Anonymous.bstrVal) };
```
Comment (tim-weis) [inline src/variant.rs:106]: Consider replacing this with

```rust
let bstr_borrow = ManuallyDrop::new(unsafe { BSTR::from_raw(vt.Anonymous.Anonymous.Anonymous.bstrVal) });
Variant::Sting(bstr_borrow.to_string())
```

This creates a non-owning `BSTR` from the get-go, so you do not have to remember and cannot forget to manually release ownership later.

**Pull request merged**